src/makeCalculator.js

```diff
@@ -3,8 +3,37 @@
 /**
  * @return {object}
  */
-function makeCalculator() {
-  // write code here
+function makeCalculator(operation, x) {
+  const calculator = {
+    result: 0,
+    add(num) {
+      this.result += num;
+    },
+    subtract(num) {
+      this.result -= num;
+    },
+    multiply(num) {
+      this.result *= num;
+    },
+    divide(num) {
+      if (num !== 0) {
+        this.result /= num;
+      } else {
+      }
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, num) {
```

Comment:
[CODE KNOWLEDGE] The `operate` method should check if the provided `callback` is a function before attempting to call it. This is to prevent runtime errors if a non-function is passed.